### PR TITLE
[Fix] AllowedReaction code

### PR DESCRIPTION
**Pull request by 6543**
fix/optimize code from #8886

before: AllowedReaction was piped throu multible templates in the worst case
now: just exec func AllowedReaction

## Fix:
if you change reactions you are not able anymore to add it by this UI element:
Bildschirmfoto zu 2019-12-27 18-38-15

### CODE REVIEW

File: modules/templates/helper.go [hunk 1]
```diff
@@ -81,6 +81,9 @@ func NewFuncMap() []template.FuncMap {
 		"LoadTimes": func(startTime time.Time) string {
 			return fmt.Sprint(time.Since(startTime).Nanoseconds()/1e6) + "ms"
 		},
+		"AllowedReactions": func() []string {
```
Comment (zeripath) [inline modules/templates/helper.go:84]: Hmm... Maybe if we're going to do this we should just allow templates to query setting?

Comment (6543) [reply modules/templates/helper.go:84]: would be a solution ... but then we can refactor a lot :)

**Comment by 6543** [comment]
@zeripath https://github.com/go-gitea/gitea/pull/9518#discussion_r361710740 is in this case a general discusion ... but this PR fixes a bug (wich I'm searching for the issue Nr right now)

**Review by zeripath — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

File: modules/templates/helper.go (hunk 1 above)
Comment (lafriks) [reply modules/templates/helper.go:84]: I also agree that we could just name function "Settings" and use it here and later in other places where needed

**Pull request merged**